Python/pystate.c

```diff
@@ -537,28 +471,23 @@ _PyRuntimeState_ReInitThreads(_PyRuntimeState *runtime)
     // This was initially set in _PyRuntimeState_Init().
     runtime->main_thread = PyThread_get_thread_ident();
 
-    /* Force default allocator, since _PyRuntimeState_Fini() must
-       use the same allocator than this function. */
-    PyMemAllocatorEx old_alloc;
-    _PyMem_SetDefaultAllocator(PYMEM_DOMAIN_RAW, &old_alloc);
-
-    PyThread_type_lock *lockptrs[NUMLOCKS] = LOCKS_INIT(runtime);
-    int reinit_err = 0;
-    for (int i = 0; i < NUMLOCKS; i++) {
-        reinit_err += _PyThread_at_fork_reinit(lockptrs[i]);
-    }
-
-    PyMem_SetAllocator(PYMEM_DOMAIN_RAW, &old_alloc);
-
     // Clears the parking lot. Any waiting threads are dead. This must be
     // called before releasing any locks that use the parking lot.
     _PyParkingLot_AfterFork();
 
+    // Re-initialize global locks
+    runtime->interpreters.mutex = (PyMutex){0};
+    runtime->xi.registry.mutex = (PyMutex){0};
+    runtime->unicode_state.ids.mutex = (PyMutex){0};
+    runtime->imports.extensions.mutex = (PyMutex){0};
+    runtime->ceval.pending_mainthread.mutex = (PyMutex){0};
+    runtime->atexit.mutex = (PyMutex){0};
+    runtime->audit_hooks.mutex = (PyMutex){0};
+    runtime->allocators.mutex = (PyMutex){0};
```

Comment:
FWIW, I liked having all the locks spelled out in one place via `LOCKS_INIT()`.  It made it much easier to add more locks.  However, this is fine since we only have the one place now where we mess with all the locks together.

Reply:
I've re-added `LOCKS_INIT()`, but I hope this is the only place we need to enumerate them.

Reply:
Thanks!